packages/pubsub/src/subscription.js

```diff
@@ -592,30 +350,154 @@ Subscription.prototype.decorateMessage_ = function(message) {
  *   var apiResponse = data[0];
  * });
  */
-Subscription.prototype.delete = function(callback) {
+Subscription.prototype.delete = function(gaxOpts, callback) {
   var self = this;
 
-  callback = callback || common.util.noop;
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
 
-  var protoOpts = {
-    service: 'Subscriber',
-    method: 'deleteSubscription'
+  var reqOpts = {
+    subscription: this.name
   };
 
+  this.request({
+    client: 'subscriberClient',
+    method: 'deleteSubscription',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, function(err, resp) {
+    if (!err) {
+      self.closeConnection_();
```

Comment:
> Why is it going to throw an error? The code will make an RPC to Pub/Sub. Then the code will asynchronously receive an error response from the service sometime later.

Sorry, let me be more specific, we will emit any errors received via `error` events. 

> It seems like the transition from 0->1 message listeners should trigger an openConnection and the transition from 1->0 message listeners should trigger a closeConnection... and nothing else. Then we preserve a straightforward invariant of "there's an open connection if and only if there are message listeners."

So what do you want to happen when the subscription is deleted? Obviously a connection shouldn't be able to be established, however it doesn't make sense to automatically remove error listeners since that is the only way we can inform the user that an error occurred.